controllers/operator_controller_test.go

```diff
@@ -348,8 +348,31 @@ var _ = Describe("Reconcile Test", func() {
 				Expect(err).NotTo(HaveOccurred())
 			})
 
+			It("verify operator status when bundle deployment status is stale while being created", func() {
+				res, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: opKey})
+				Expect(res).To(Equal(ctrl.Result{}))
+				Expect(err).NotTo(HaveOccurred())
+
+				By("fetching the updated operator after reconcile")
+				op := &operatorsv1alpha1.Operator{}
+				err = cl.Get(ctx, opKey, op)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("checking the expected conditions")
+				cond := apimeta.FindStatusCondition(op.Status.Conditions, operatorsv1alpha1.TypeReady)
+				Expect(cond).NotTo(BeNil())
+				Expect(cond.Status).To(Equal(metav1.ConditionUnknown))
+				Expect(cond.Reason).To(Equal(operatorsv1alpha1.ReasonInstallationStatusUnknown))
+				Expect(cond.Message).To(Equal(fmt.Sprintf("waiting for bundleDeployment %q status to be updated. BundleDeployment conditions out of date.", bd.Name)))
+			})
+
 			It("verify operator status when bundle deployment is waiting to be created", func() {
 				By("running reconcile")
+				bd.Status.ObservedGeneration = bd.GetGeneration()
```

Comment:
This is hacky, but needed since we are trying to mimic >1 reconcile.

Reply:
A better approach might be to have separate `When` blocks. Something along these lines, maybe?

```go
When("The bundle deployment status is stale", func() {})
When("The bundle deployment status is up-to-date", func() {
	BeforeEach(func() {
		bd.Status.ObservedGeneration = bd.GetGeneration()
		By("updating the status of bundleDeployment")
		err := cl.Status().Update(ctx, bd)
		Expect(err).NotTo(HaveOccurred())
	}
})
```

Also, I find it easier to think about these tests not as mimicking multiple reconciles, but more like "what happens during a given Reconcile when the state of the world is X?". And then we would ideally have tests for all the various X's such that we get really good code coverage in `Reconcile`.

Reply:
Made the change, just left the status update call in `It` blocks, so that we update the status only once after setting the right conditions.